Sming/Services/libemqtt/libemqtt.c

```diff
@@ -400,13 +400,13 @@ int mqtt_ping(mqtt_broker_handle_t* broker) {
 	return 1;
 }
 
-int mqtt_publish(mqtt_broker_handle_t* broker, const char* topic, const char* msg, uint8_t retain) {
-	return mqtt_publish_with_qos(broker, topic, msg, retain, 0, NULL);
+int mqtt_publish(mqtt_broker_handle_t* broker, const char* topic, const char* msg, uint16_t mlength, uint8_t retain) {
+	return mqtt_publish_with_qos(broker, topic, msg, mlength, retain, 0, NULL);
 }
 
-int mqtt_publish_with_qos(mqtt_broker_handle_t* broker, const char* topic, const char* msg, uint8_t retain, uint8_t qos, uint16_t* message_id) {
+int mqtt_publish_with_qos(mqtt_broker_handle_t* broker, const char* topic, const char* msg, uint16_t mlength, uint8_t retain, uint8_t qos, uint16_t* message_id) {
 	uint16_t topiclen = strlen(topic);
-	uint16_t msglen = strlen(msg);
+	uint16_t msglen = mlength;
```

Comment:
@MayaPosch Please, remove line 409 completely and instead name the function parameter msglen. Apply the name change also to the header file.